### ISSUE TITLE
False positive for nested selectors within placeholder selectors in selector-no-qualifying-type

### ISSUE THREAD

**Issue by cleavera** [stylelint/stylelint#2041]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

A bug where when running selector-no-qualifying-type as true, nested rules within abstract class are returning as a linting error even if they are used within classes.

> Which rule, if any, is this issue related to?

selector-no-qualifying-type

> What CSS is needed to reproduce this issue?

```scss
%MyAbstractClass {
  &.MyNestedClass { // Linting error reported here
    // Some declarations

    &.AnotherNestedClass { // Linting error reported here
      // Some declarations
    }
  } 
}

.MyOuterClass {
  @extend %MyAbstractSelector;
}
```

> What stylelint configuration is needed to reproduce this issue?

```json
{
  "rules": {
    "selector-no-qualifying-type": true
  }
}
```

> Which version of stylelint are you using?

`7.5.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

SCSS

> What did you expect to happen?

It treats the abstract class as an element selector and report linting errors on any nested classes. I would expect these not to throw linting errors unless the abstract class is extended within an element selector.

> What actually happened (e.g. what warnings or errors you are getting)?

```shell
test.scss
 2:3  ×  Unexpected qualifying type selector   selector-no-qualifying-type
 5:5  ×  Unexpected qualifying type selector   selector-no-qualifying-type
```

**Comment by evilebottnawi** [comment]
@cleavera format your issue as a template

**Comment by cleavera** [comment]
@evilebottnawi Done :)

**Comment by evilebottnawi** [comment]
@cleavera Thanks

**Comment by jeddy3** [comment]
>> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

> No

Can you confirm what syntax you're using? As this doesn't look like standard syntax. If it isn't standard syntax, can you point us towards a resource for the `%...` construct please?

**Comment by cleavera** [comment]
Ooops, sorry, it's SCSS

**Comment by jeddy3** [comment]
Thanks. I was able to replicate the issue.

It looks to be an issue with SCSS placeholder selectors _when combined with_ nested selectors.

```scss
%placeholder {
  &.class {} /* "2:3 Unexpected qualifying type selector" */
}

%placeholder.class {} /* no warning */
```

This is odd as the rule _resolves nested selectors_ before running any checks and so both statements above should behave in the same way, but they do not. This is likely where the bug resides.

@cleavera Thanks for the original report. Please consider [helping out](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#fixing-bugs) and investigating the issue some more, and hopefully proposing a fix via a pull request.

I'll label the issue accordingly.

**Comment by evilebottnawi** [comment]
@jeddy3 i am fix this today :smile:

**Comment by davidtheclark** [comment]
Closed by #2044.
